tests/unit/data_access/test_entity_queries.py

```diff
@@ -1,9 +1,111 @@
 from sqlalchemy.orm import Query
-from application.data_access.entity_queries import _apply_limit_and_pagination_filters
+from application.data_access.entity_queries import (
+    _apply_limit_and_pagination_filters,
+    get_entity_search,
+)
+from application.db.session import get_context_session
 from application.db.models import EntityOrm
 
 
 def test__apply_limit_and_pagination_filters_with_no_filters_applied():
     query = Query(EntityOrm)
     result = _apply_limit_and_pagination_filters(query, params={"dataset": "testing"})
     assert result._limit_clause is None
+
+
+dataset_name = "test-dataset"
+base_id = 1000
+
+
+@pytest.fixture
+def remove_entities():
+    with get_context_session() as session:
+        session.query(EntityOrm).filter(EntityOrm.dataset == dataset_name).delete()
+        session.commit()
+
+
+@pytest.fixture
+def create_entities():
+    """
+    Create a dataset with
+    + a discontinuous series of IDs (entity column)
+    + and the rows not all inserted in increasing order
+    """
+
+    with get_context_session() as session:
+
+        for num in range(1, 24):
+            e1 = EntityOrm(
+                entity=base_id + num, name=f"S01Entity{num+1:04}", dataset=dataset_name
+            )
+            session.add(e1)
+
+        for num in range(73, 95):
+            e2 = EntityOrm(
+                entity=base_id + num, name=f"S02Entity{num+1:04}", dataset=dataset_name
+            )
+            session.add(e2)
+
+        for num in range(43, 55):
+            e3 = EntityOrm(
+                entity=base_id + num, name=f"S03Entity{num+1:04}", dataset=dataset_name
+            )
+            session.add(e3)
+
+        session.commit()
+
+
+def test__apply_limit_and_pagination_filters_only(
+    test_data, remove_entities, create_entities
+):
+    """
+    Create a dataset, and page through it, gathering the entity values (IDs) for each row.
+    Make sure no values are seen twice.
+    Make sure all values in db are actually collected in paged data.
+    """
+    # tidy()
+    # setupDB()
+
+    allSeen = []  # Add the entities we have seen whilst paging through the dataset
+    last = 0  # tracks the last entity id we got. Used in get_entity_search below.
+
+    search = get_entity_search(
+        parameters={"dataset": [dataset_name], "offset": last, "limit": 10}
+    )
+    entities = search.get(
+        "entities"
+    )  # there are also count and params in the returned search object.
+
+    while len(entities) == 10:
+        search = get_entity_search(
+            parameters={"dataset": [dataset_name], "offset": last, "limit": 10}
+        )
+        entities = search.get("entities")
+
+        newValues = [x.entity for x in entities]
+
+        # Make sure no values are seen twice.
+        for value in newValues:
+            assert value not in allSeen
+
+        allSeen = allSeen + newValues
+        assert len(allSeen) > 0
+
+        allSeen.sort()
+
+        last = entities[-1].entity
+
```

Comment:
I think because your offsetting by the last entity your not going to get the issue I've mentioned above. you offset should increase by ten each time. This then means that your not really using an offset in the same sense that the search.html page expects it. 

Changing this test to last = last +10 causes it to fail implying that using a sequence of offsets like 0,10,20 or 0,5,10 would fail as we'd see the same entities